modules/sor/sorV2/lib/pools/fx/fxPool.ts

```diff
@@ -231,4 +231,38 @@ export class FxPool implements BasePool {
             swapKind,
         };
     }
+
+    addLiquiditySingleTokenExactIn(tokenIn: Token, bpt: Token, amount: TokenAmount): TokenAmount {
+        throw new Error('FX - addLiquiditiySingleTokenExactIn: Invalid Method');
```

Comment:
As I mentioned, all pools should have all add/remove liquidity types available out of the box on v3.
I'm not really sure how you're going to handle that in the context of replicating pool math, since these methods were not available in v2 for these pools.
We'll likely have to replicate the `invariant` solution implemented by the SC team.